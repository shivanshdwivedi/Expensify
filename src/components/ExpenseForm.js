// jshint esversion : 6
// jshint esversion : 9


import React from 'react';

export default class ExpenseForm extends React.Component {
      state = {
        description: '',
        note: ''
      };
      onDescriptionChange = (e) => {
      const description = e.target.value;
      this.setState(() => ({ description }));
      };

      onNoteChange = (e) => {
        const note = e.target.value;
        this.setState(() => ({ note }));
      }

    render() {
        return (
            <div>
                Expense Form
                <form>
                    <input
                    type = 'text'
                    placeholder = 'Description'
                    autofocus
                    value = {this.state.description}
                    onChange = {this.onDescriptionChange}
                    />
                    <input
                    type = 'number'
                    placeholder = 'Amount' 
                    />
                    <div>
                    <textarea
                    placeholder = 'Add a note for your expense(optional)'
                    value = {this.state.note}
                    onChange = {this.onNoteChange}
                    />
                    </div>
                    
                    <div>
                    <button>Add Expense</button>
                    </div>
                    

                </form>
            </div>
        )
    }
}