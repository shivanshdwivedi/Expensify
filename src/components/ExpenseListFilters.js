// jshint esversion : 6
// jshint esversion : 9

import React from 'react';
import {connect} from 'react-redux';
import { setTextFilter, setEndDate, setStartDate } from '../actions/filters';
import { sortByDate } from '../actions/filters';
import { sortByAmount , startDate ,endDate} from '../actions/filters';
import {DateRangePicker} from 'react-dates';

class ExpenseListFilters extends React.Component{
    state = {
        calenderFocused: null
    };
    onDatesChange = ({startDate , endDate}) => {
     this.props.dispatch(setStartDate(startDate));
     this.props.dispatch(setEndDate(endDate));
    };
    onFocusChange = (calenderFocused) => {
    this.setState(() => ({calenderFocused}));
    }
    render(){
        return(
            <div>
                <input type = 'text' value = {this.props.filters.text} onChange = {(e) => {
                    props.dispatch(setTextFilter(e.target.value))
                }}/>
                <select 
                    value = {this.props.filters.sortBy}  
                    onChange = {(e) =>{
                    if(e.target.value === 'date'){
                        props.dispatch(sortByDate());
                    }else if(e.target.value === 'amount'){
                     props.dispatch(sortByAmount());
                    }
                     }}
                >
                     <option value = 'amount'>Amount</option>
                    <option value = 'date'>Date</option>
                </select>
              <DateRangePicker 
                  startDate = {this.props.filters.startDate}
                  endDate = {this.props.filters.endDate}
                  onDatesChange= {this.onDatesChange}
                  focusedInput = {this.state.calenderFocused}
                  onFocusChange = {this.onFocusChange}
                  showClearDates = {true}
                  numberOfMonths = {1}
                  isOutsideRange = {() => false}
              />

            </div>
        )

    }
}

const mapStateToProps = (state) => {
    return {
        filters: state.filters
    }
}

export default connect(mapStateToProps)(ExpenseListFilters);