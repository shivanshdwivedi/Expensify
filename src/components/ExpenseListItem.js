// Export a stateless functional component
// description , amout , createdAt
import React from 'react';
import { Link } from 'react-router-dom';
import numeral from 'numeral';
import moment from 'moment';

const ExpenseListItem = ({description , amount , createdAt ,  id }) => (
    <div>
    <Link to ={`/edit/${id}`}>   
    <h3> {description} </h3>
    </Link>
        <p>
        {numeral(amount/100).format('Rs0,0.00')} 
        - 
        {moment(createdAt).format('MMMM Do , YYYY')}
        </p>
    </div>
);

export default ExpenseListItem;