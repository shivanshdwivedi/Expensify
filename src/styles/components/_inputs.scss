.text-input , .select{
    border: 1px solid #cacccd;
    height: 50;
    font-size: $font-size-large;
    font-weight: 300;
    padding: $s-size;
}

.select {
    @extend .text-input;;
}

.textarea {
    @extend .text-input;
    height: 10rem;
}