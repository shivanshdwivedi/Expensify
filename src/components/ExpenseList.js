import React from 'react';
import {connect} from 'react-redux';
import ExpenseListItem from './ExpenseListItem';
const ExpenseList = (props) => (
    <div>
        <h1>
            Expense List
        </h1>
            {ExpenseListItem}
    </div>
);

const mapStateToProps = (state) => {
  return{
   expenses: state.expenses,
   filters: state.filters
  };
};

export default connect(mapStateToProps)(ExpenseList)

