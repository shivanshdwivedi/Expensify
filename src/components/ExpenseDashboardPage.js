// jshint esversion : 6
// jshint esversion : 9

import React from 'react';
import ExpenseList  from './ExpenseList';
import ExpenseListFilters from './ExpenseListFilters';

const ExpenseDashboardPage = () => (
    <div>
       <ExpenseListFilters />
       <ExpenseList />
    </div>
);

export default ExpenseDashboardPage;