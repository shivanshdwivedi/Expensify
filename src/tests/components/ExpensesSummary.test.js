// jshint esversion : 6

import React from 'react';
import {shallow} from 'enzyme';
import {ExpensesSummary} from '../../components/ExpensesSummary';

test('' , () => {
const wrapper = shallow(<ExpensesSummary expenseCount = {1} expensesTotal = {235} />);
expect(wrapper).toMatchSnapshot();
});

test('' , () => {
const wrapper = shallow(<ExpensesSummary expenseCount = {23} expensesTotal = {23512340987} />);
expect(wrapper).toMatchSnapshot();
});
