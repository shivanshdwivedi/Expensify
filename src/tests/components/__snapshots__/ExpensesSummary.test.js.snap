// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
<div>
  <h1>
    Viewing 
    1
     
    expense
     totalling 
    235.00
  </h1>
</div>
`;

exports[` 2`] = `
<div>
  <h1>
    Viewing 
    23
     
    expenses
     totalling 
    23,512,340,987.00
  </h1>
</div>
`;
